src/main/scala/mesosphere/marathon/core/election/ElectionModule.scala

```diff
@@ -1,40 +1,42 @@
 package mesosphere.marathon
 package core.election
 
-import akka.actor.ActorSystem
+import akka.actor.{ ActorSystem, Cancellable }
 import akka.event.EventStream
-import mesosphere.marathon.core.base.{ CrashStrategy, LifecycleState }
-import mesosphere.marathon.core.election.impl.{ CuratorElectionService, PseudoElectionService }
+import akka.stream.scaladsl.Source
+import java.util.concurrent.ExecutorService
+import mesosphere.marathon.core.base.CrashStrategy
+import mesosphere.marathon.util.LifeCycledCloseable
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
 
 class ElectionModule(
     config: MarathonConf,
     system: ActorSystem,
     eventStream: EventStream,
     hostPort: String,
-    lifecycleState: LifecycleState,
-    crashStrategy: CrashStrategy) {
+    crashStrategy: CrashStrategy,
+    electionExecutor: ExecutorService
+) {
 
-  lazy val service: ElectionService = if (config.highlyAvailable()) {
+  lazy private val electionBackend: Source[LeadershipState, Cancellable] = if (config.highlyAvailable()) {
     config.leaderElectionBackend.get match {
       case Some("curator") =>
-        new CuratorElectionService(
-          config,
+        val client = new LifeCycledCloseable(CuratorElectionStream.newCuratorConnection(config))
+        sys.addShutdownHook { client.close() }
```

Comment:
Don't we have to close the stream as well?

Reply:
stream is closed implicitly by the latch closing

Reply:
(watch events fire on connection loss)